### DISCUSSION TITLE
Sliding channel #2 visualization bug

### DISCUSSION THREAD

**Discussion by voidexp** [OpenHantek/OpenHantek6022#295] [Q&A] [answered]
**Describe the bug**
Channel 2 of the 6022BL scope seems not to catch the trigger (if I use the right terminology) and continuously slides from right to left.
Check out animation.

No problems on Channel 1. Tried switching probes, problem remains.

**To Reproduce**
1. Start OpenHantek6022
2. Attach the channel 2 probe to reference 2Vpp/1kHz signal.
3. See the sliding erratic behavior.

**Expected behavior**
The signal stays fixed in place.

**Screenshots**
https://user-images.githubusercontent.com/595098/165324333-c8efecca-94a3-4bcf-983e-c5c489086414.mp4

**Computer environment (please complete the following information):**
 - OpenHantek version: *stable release 3.2.5 31c5ee2*
 - OS: Windows 10 Version 10.0.22000 Build 22000
 - Video hardware: nVidia 1080 Ti, driver 512.15
 - OpenGL version: OpenGL 4.6.0 

**Scope device (please complete the following information):**
 - Device *Hantek 6022BL*
 - Program top line  *OpenHantek6022 (3.2.5) - Device SDO-6022BL (FW 209)*
 - Input signal: *calibration out 1 kHz, 2Vpp*
 - Probe setting:  *probe X1*

**Additional context**
Calibrated using the HantekAPI calibration script and firmware, version 2.09

**Comment by Ho-Ro** [comment, marked as answer]
> Expected behavior
The signal stays fixed in place.

It would trigger on CH2 if you had selected this channel as the trigger source. However, your source is set to CH1:
![image](https://user-images.githubusercontent.com/12542359/165698727-44299459-78b9-48be-8ef5-21dae14992dd.png)
![image](https://user-images.githubusercontent.com/12542359/165700089-7acb5a6e-bb5f-43dd-bc4c-7a612ce0d408.png)

This can also be seen from the colour of the trigger position slider at the top of the screen: It is yellow for CH1 and blue for CH2:
![image](https://user-images.githubusercontent.com/12542359/165698616-dd869b58-5a95-4857-b58a-56f0a19d75c4.png)

**Comment by voidexp** [comment]
Oh... stupid me :) But wouldn't that leave the first channel floating then? What if I want to compare two signals at once?

**Comment by Ho-Ro** [comment]
If both channels are unrelated, only one channel can be fixed. If you want to see a steady picture, just use single trigger mode or stop the capturing with the pause button.
